src/telem/lib/input/mekanism/GasGeneratorInputAdapter.lua

```diff
@@ -0,0 +1,70 @@
+local o = require 'telem.lib.ObjectModel'
+local t = require 'telem.lib.util'
+
+local InputAdapter      = require 'telem.lib.InputAdapter'
+local Metric            = require 'telem.lib.Metric'
+local MetricCollection  = require 'telem.lib.MetricCollection'
+
+local GasGeneratorInputAdapter = o.class(InputAdapter)
+GasGeneratorInputAdapter.type = 'GasGeneratorInputAdapter'
+
+function GasGeneratorInputAdapter:constructor (peripheralName, categories)
+    self:super('constructor')
+
+    -- TODO this will be a configurable feature later
+    self.prefix = 'mekgasgen:'
+
+    -- TODO make these constants
+    local allCategories = {
+        'basic',
+    }
+
+    if not categories then
+        self.categories = { 'basic' }
+    elseif categories == '*' then
+        self.categories = allCategories
+    else
+        self.categories = categories
+    end
+
+    -- boot components
+    self:setBoot(function ()
+        self.components = {}
+
+        self:addComponentByPeripheralID(peripheralName)
+    end)()
+end
+
+function GasGeneratorInputAdapter:read ()
+    self:boot()
+
+    local source, generator = next(self.components)
+
+    local metrics = MetricCollection()
+
+    local loaded = {}
+
+    for _,v in ipairs(self.categories) do
+        -- skip, already loaded
+        if loaded[v] then
+            -- do nothing
+
+        -- Literally all we have lmao
+        elseif v == 'basic' then
```

Comment:
Categories should be expanded:

Basic:
- energy_filled_percentage
- fuel_filled_percentage
- production_rate

Fuel:
- fuel_amount
- fuel_capacity
- fuel_needed

Energy:
- energy